weather_dl/download_pipeline/manifest.py

```diff
@@ -242,22 +447,138 @@ def _update(self, download_status: DownloadStatus) -> None:
                 logger.debug(download_status)
 
 
+class BQManifest(Manifest):
+    """Writes a JSON representation of the manifest to BQ file.
+
+    This is an append-only implementation, the latest value in the manifest
+    represents the current state of a download.
+    """
+    def __init__(self, location: Location) -> None:
+        super().__init__(Location(location[5:]))
+        TABLE_SCHEMA = [
+            bigquery.SchemaField('config_name', 'STRING', mode='REQUIRED',
+                                 description="The name of the config file associated with the request."),
+            bigquery.SchemaField('selection', 'JSON', mode='REQUIRED',
+                                 description="Copy of selection section of the configuration."),
+            bigquery.SchemaField('location', 'STRING', mode='REQUIRED',
+                                 description="Location of the downloaded data."),
+            bigquery.SchemaField('stage', 'STRING', mode='NULLABLE',
+                                 description="Current stage of request : 'fetch', 'download', 'retrieve', 'upload' "
+                                 "or None."),
+            bigquery.SchemaField('status', 'STRING', mode='REQUIRED',
+                                 description="Download status: 'scheduled', 'in-progress', 'success', or 'failure'."),
+            bigquery.SchemaField('error', 'STRING', mode='NULLABLE',
+                                 description="Cause of error, if any."),
+            bigquery.SchemaField('user', 'STRING', mode='REQUIRED',
+                                 description="Identifier for the user running the download."),
+            bigquery.SchemaField('size', 'FLOAT', mode='NULLABLE',
+                                 description="File size in GB."),
+            bigquery.SchemaField('scheduled_time', 'TIMESTAMP', mode='NULLABLE',
+                                 description="A UTC datetime when download was scheduled."),
```

Comment:
Another idea from Tyler + I: Let's include a nullable Geospatial Polygon. This column would represent the `area` lat/lngs from the selection. If we record the area covered by a shard, we can use BQ to quickly check if we have data coverage. 

If the `area` is omitted, let's assume it has global coverage (put in a default global polygon).

Reply:
Thanks Rahul :)

Reply:
Done. Included the `area` (GeoJSON geometry object) as STRING in BQ schema.

Some useful functions for analysis purpose:
* [ST_GEOGFROMGEOJSON](https://cloud.google.com/bigquery/docs/reference/standard-sql/geography_functions#st_geogfromgeojson) : To convert a GeoJSON geometry object into a GEOGRAPHY value. 
* ST_COVERS(geography_expression, ST_GEOGPOINT(longitude, latitude)): To check if a point lies in the given area or not.

Reply:
Looks good to me!